File: backend/app/schemas/capacity.py
from enum import Enum

from pydantic import BaseModel


class CapacityStatus(str, Enum):
    AVAILABLE = "AVAILABLE"
    FULLY_ALLOCATED = "FULLY_ALLOCATED"
    OVERALLOCATED = "OVERALLOCATED"


class EmployeeCapacityRead(BaseModel):
    employee_id: int
    employee_name: str
    allocated_percent: int
    available_percent: int
    status: CapacityStatus

File: frontend/src/types/capacity.ts
export type CapacityStatus = 'AVAILABLE' | 'FULLY_ALLOCATED' | 'OVERALLOCATED'

export interface EmployeeCapacity {
  employee_id: number
  employee_name: string
  allocated_percent: number
  available_percent: number
  status: CapacityStatus
}
